src/lib.rs

```diff
@@ -0,0 +1,289 @@
+#![crate_type = "staticlib"]
+
+use libc::{c_char, c_int, c_void, size_t, ssize_t};
+use std::ffi::CStr;
+use std::io::ErrorKind::ConnectionAborted;
+use std::io::{Cursor, Read, Write};
+use std::slice;
+use std::sync::Arc;
+
+use rustls::{ClientSession, Session, ALL_CIPHERSUITES};
+
+type CrustlsResult = c_int;
+
+pub const CRUSTLS_OK: c_int = 0;
+pub const CRUSTLS_ERROR: c_int = 1;
+
+static mut RUSTLS_CONFIG: Option<Arc<rustls::ClientConfig>> = None;
+
+#[no_mangle]
```

Comment:
I'd strongly encourage you not to include global state, it makes things incredibly painful when multiple dependencies want to use this in  single project.